paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass.cc

```diff
@@ -426,9 +427,25 @@ void CPUQuantizeSquashPass::MultipleQuantizeSquash(Graph* graph) const {
         auto quant_out = quant_op->outputs[0];
         auto last_op = quant_out->outputs[0];
         auto last_op_op = last_op->Op();
+        bool residual_fc = false;
 
         std::string last_op_input_name =
             FindInputNameByVarName(last_op_op, quant_out->Name());
+        if (last_op_input_name.empty()) {
+          last_op_input_name =
+              FindOutputNameByVarName(last_op_op, quant_out->Name());
+          PADDLE_ENFORCE_EQ(last_op_input_name,
+                            "ResidualData",
+                            platform::errors::InvalidArgument(
+                                "Only the ResidualData output is allowed to be "
+                                "linked to earlier op."));
+          PADDLE_ENFORCE_EQ(last_op_op->Type(),
+                            "fc",
+                            platform::errors::InvalidArgument(
+                                "Only fc operator supports ResidualData output "
+                                "linked as input."));
+          residual_fc = true;
+        }
```

Comment:
This logic seems a little bit odd for me.
Does residual connection must have an empty name?
Are there any other scenarios where the name can be empty?
What if op will have empty name in graph, will this pass throw an exception?

Reply:
it's empty because FindInputNameByVarName returns empty string if it's not found. In that case we search for name in the outputs instead of inputs because fc is special case when residualData is output instead of input.